src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/QueryStringValueProvider.cs

```diff
@@ -86,6 +86,11 @@ public override ValueProviderResult GetValue(string key)
                 throw new ArgumentNullException(nameof(key));
             }
 
+            if (key.Length == 0)
```

Comment:
This needs an explanation as to why in the comments 😆 

Should we also consider changes to the other value providers?

Reply:
Considering this is going in 3.0, I find it more palatable to make it for all our value providers. Makes the behavior more predictable.

Reply:
Yeah I guess I was wondering if it's possible for our other value providers. Like, we should find out if `Form` is hardened against empty keys, n, stuff like that.

Reply:
Is there an answer to this cool question?

Reply:
This change hardens the following value providers: `FormValueProvider`, `QueryStringValueProvider`, `RouteValueProvider`. I left the jquery providers as is. I didn't want to spend time trying to see what the behavior is across versions.